src/types/string.rs

```diff
@@ -172,6 +185,20 @@ impl<'source> FromPyObject<'source> for String {
     }
 }
 
+impl<'source> FromPyObject<'source> for char {
+    fn extract(obj: &'source PyAny) -> PyResult<Self> {
```

Comment:
Similarly can skip the lifetime here:

```suggestion
impl FromPyObject<'_> for char {
    fn extract(obj: &PyAny) -> PyResult<Self> {
```